### ISSUE TITLE
Directives should always be passed using brackets 

### ISSUE THREAD

**Issue by luiscla27** [eslint/eslint#14326]
This rule proposal is related to this issue, the new rule would be readed like this:
_Every directive declared in the current NgModule, should always be used using brackets_

**What category of rule is this? (place an "X" next to just one item)**

- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (promotes good practices)

**Provide 2-3 code examples that this rule will warn about:**

```html
<button SomeCustomDirective="5" />
```

The expected would be this:
```html
<button [SomeCustomDirective]="5" />
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Without the brackets the IDE doesn't have a way of knowing which variable type is expecting, so it's easy to sometimes pass a wrong value... specially when the directive is refactored

In my case, our team separated a bunch of angular components, each one to a single own module so we were able to call them lazy. And we didn't realized that there it was a lot of missing module imports from NgModule's until we ran the app. I think our case counts as refactorization.

**Are you willing to submit a pull request to implement this rule?**
No, I have a full time job so I don't think so. Also, this is not an easy one... As it should look for existing directives declared at the current module.

**Comment by nzakas** [comment]
ESLint core rules only relate to JavaScript, not any specific framework. So this proposal doesn’t fit our criteria. I’d suggest looking for an Angular-specific ESLint plugin to propose this rule for.

**Comment by luiscla27** [comment]
Thank you for your feedback
